projects/arlas-mapbox/src/lib/legend/legend.service.ts

```diff
@@ -0,0 +1,356 @@
+/*
+ * Licensed to Gisaïa under one or more contributor
+ * license agreements. See the NOTICE.txt file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Gisaïa licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Injectable } from '@angular/core';
+import { CircleLegend, getMax, Legend, LegendData, PROPERTY_SELECTOR_SOURCE } from 'arlas-map';
+import { TranslateService } from '@ngx-translate/core';
+import { CirclePaint, FillPaint, HeatmapPaint, Layer, LinePaint, SymbolPaint } from 'mapbox-gl';
+import { HistogramData } from 'arlas-d3/histograms/utils/HistogramUtils';
+import { MAX_CIRLE_RADIUS } from 'arlas-map';
+import { LineLegend } from 'arlas-map';
+import { MAX_LINE_WIDTH } from 'arlas-map';
+import { LayerMetadata } from 'arlas-map';
+import { FillLegend, HeatmapLegend } from 'arlas-map';
+import { LabelLegend, LegendService } from 'arlas-map';
+import * as tinycolor from 'tinycolor2';
```

Comment:
should be import tinycolor from 'tinycolor2'

Reply:
fixed